Add plugin: SVG Style Editor

arg1998 opened the pull request:
# I am submitting a new Community Plugin

Description from the original repository:
 > Obsidian SVG Styler is a simple plugin designed to help you easily modify the style properties of SVG images directly within the editor. With this plugin, you can customize properties like color, border, opacity, and more. Supported drawable tags include: `path`, `line`, `rect`, `circle`, `ellipse`, and `polygon`.

## Repo URL

<!--- Paste a link to your repo here for easy access -->
[arg1998/obsidian-svg-styler](https://github.com/arg1998/obsidian-svg-styler)

## Release Checklist
- [x] I have tested the plugin on
  - [x]  Windows
  - [x]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[# Obsidian SVG Style Editor](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/README.md#L1-L2), [Obsidian SVG Styler](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/README.md#L3)
Naming something "Obsidian xyz" is reserved for first party products we create, please change this header.

[const svgson = require("svgson");](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L19-L20)
Please stick to one import style

[id: "obsidian-svg-styler",](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L55)
The command ID should describe what the command does, and should not contain the plugin id.

[callback: () => {](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L57)
This should be a [conditional command](https://docs.obsidian.md/Plugins/User+interface/Commands#Conditional+commands)

[const data = await readFile(this.getDataFilePath(), "utf8");](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L106), [await writeFile(this.getDataFilePath(), data, "utf8");](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L122)
Please use `loadData()` and `saveData()` from the Obsidian API for storing your data.
This allows the data to be compatible with syncing platforms like Obsidian Sync, an example can be found in the sample plugin.

["You can Change the default resource path in](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L152),[.setName("Duplicate SVG Location")](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L167), [.setName("Custom Directory Path")](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L183)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[containerEl.createEl("h1", { text: "SVG Styler Settings" });](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L155)
[Don't add a top-level heading in the settings tab, such as "General", "Settings", or the name of your plugin.](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Only+use+headings+under+settings+if+you+have+more+than+one+section.)

[containerEl.createEl("h3", { text: "Duplicate SVG Directory" });](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L158)
For section headings in settings use:
```ts
new Setting(containerEl).setName('name here').setHeading();
```

[// Add a "Save" button](https://github.com/arg1998/obsidian-svg-styler/blob/28c4037fa566d298f76c6a6b872e8f1eb3f6547c/main.ts#L196-L229)
A settings tab should not have a "Save" button, settings are saved right after they are changed.

arg1998 commented:
@joethei Thanks for the thorough review. I addressed all of the issues mentioned above. You can see the changelog for the latest [release](https://github.com/arg1998/obsidian-svg-styler/releases/tag/1.1.2) here: https://github.com/arg1998/obsidian-svg-styler/compare/1.1.1...1.1.2  .


I have one issue remaining. Can you tell me why `this.editor.refresh()` doesn't update the current open editor in my [`refreshEditor()`](https://github.com/arg1998/obsidian-svg-styler/blob/2660dce25cfe1b936d954516b300a0243d08628e/main.ts#L688) method? I am calling `refresh()` method on the `editor` object passed by the `editorCheckCallback: (checking: boolean, editor: Editor, view: MarkdownView) => {}`.  The workaround works perfectly and re-paints the editor.

arg1998 commented:
@joethei

I have fixed the plugin id as you mentioned. Also resolved the merged conflicts with upstream. Hwoever, the `community-plugin.json` needs some formatting.
